### PR TITLE
4.2 Added multiple winners

### PR DESCRIPTION
**Pull request by Kayanski**
This PR aims at resolving task 4.2

Multiple Raffle Winners( Urgent)
Implement support to define the number of winners for a raffle if multiple assets are provided.

Depends on https://github.com/AtlasDA0/app-contracts/pull/22

Still needs TESTING

### CODE REVIEW

File: contracts/raffles/src/utils.rs [hunk 1]
```diff
@@ -109,45 +109,61 @@ pub fn get_raffle_winner(
...
-    // We pick a winner id
-    let winner_id = int_in_range(randomness, 0, raffle_info.number_of_tickets);
-    let winner = RAFFLE_TICKETS.load(deps.storage, (raffle_id, winner_id))?;
+    let nb_winners = if raffle_info.raffle_options.one_winner_per_asset {
+        raffle_info.assets.len()
+    } else {
+        1
+    };
+
+    let winner_ids = ints_in_range(randomness, nb_winners, 0, raffle_info.number_of_tickets - 1);
```
Comment (kakucodes) [inline contracts/raffles/src/utils.rs]: do we know if `ints_in_range` is able to give repeated numbers or are they unique? either way does this match the business logic we're looking for? should a raffle entrant be able to win multiple items with one ticket?

Comment (Kayanski) [reply contracts/raffles/src/utils.rs]: It's possible there is repetitions. I think that's a good behavior

Comment (Kayanski) [reply contracts/raffles/src/utils.rs]: @Jetblaster do you agree withi this current logic ? We are saying that a single ticket could win multiple NFTs in this case

Comment (Jetblaster) [reply contracts/raffles/src/utils.rs]: Can we keep booth option open. The raffle creator can pick if he want this option open.

Comment (Kayanski) [reply contracts/raffles/src/utils.rs]: Yes !

Comment (Jetblaster) [reply contracts/raffles/src/utils.rs]: Is this correct 

1st option:
- Each raffle ticket purchase grants users a chance to win one NFT. It doesn't matter how many tickets they buy; they can only win one NFT.

2nd option:
- Each raffle will provide one chance to win.
- By purchasing multiple raffle tickets, you also increase your chances of winning multiple times.

Comment (Kayanski) [reply contracts/raffles/src/utils.rs]: Yes

Comment (Kayanski) [reply contracts/raffles/src/utils.rs]: I think we will go with a mix of the 2 options --> Each ticket has a right to win 1 NFT maximum.
So one buyer can win multiple NFTs but only with multiple tickets

File: contracts/raffles/src/state.rs [hunk 2]
```diff
@@ -130,6 +130,7 @@ pub struct RaffleOptions {
     pub max_ticket_per_address: Option<u32>, // max amount of tickets able to bought per address
     pub raffle_preview: u32,               // ?
     pub min_ticket_number: Option<u32>,    // Minimum ticket number for a raffle to close.
+    pub one_winner_per_asset: bool, // Allows to set multiple winners per raffle (one per asset)
```
Comment (kakucodes) [inline contracts/raffles/src/state.rs:133]: I do wonder if we're due for some refactoring the `RaffleOptions` to attempt to ensure that impossible states are impossible to represent. but without anything particularly concrete to recommend i'm going to approve the changes in the pr.

**Review by kakucodes — APPROVED** [review]

File: contracts/raffles/src/utils.rs (hunk 1 above)
Comment (Jetblaster) [reply contracts/raffles/src/utils.rs]: Sound good

**Pull request merged**